transports/bifrost-http/main.go

```diff
@@ -1,60 +1,69 @@
 // Package http provides an HTTP service using FastHTTP that exposes endpoints
 // for text and chat completions using various AI model providers (OpenAI, Anthropic, Bedrock, etc.).
-
+//
 // The HTTP service provides two main endpoints:
 //   - /v1/text/completions: For text completion requests
 //   - /v1/chat/completions: For chat completion requests
-
+//
 // Configuration is handled through a JSON config file and environment variables:
 //   - Use -config flag to specify the config file location
 //   - Use -env flag to specify the .env file location
 //   - Use -port flag to specify the server port (default: 8080)
 //   - Use -pool-size flag to specify the initial connection pool size (default: 300)
-
-// try running the server with:
-// go run http.go -config config.example.json -env .env -port 8080 -pool-size 300
-// after setting the environment variables present in config.example.json in your .env file.
+//
+// Example usage:
+//   go run http.go -config config.example.json -env .env -port 8080 -pool-size 300
+//   after setting the environment variables present in config.example.json in your .env file.
 
 package main
 
 import (
 	"encoding/json"
-	"errors"
 	"flag"
 	"fmt"
 	"log"
-	"os"
-	"reflect"
 	"strings"
-	"sync"
 
 	"github.com/fasthttp/router"
-	"github.com/joho/godotenv"
 	bifrost "github.com/maximhq/bifrost/core"
 	schemas "github.com/maximhq/bifrost/core/schemas"
-	"github.com/maximhq/bifrost/core/schemas/meta"
 	"github.com/maximhq/bifrost/transports/bifrost-http/integrations"
 	"github.com/maximhq/bifrost/transports/bifrost-http/integrations/genai"
+	"github.com/maximhq/bifrost/transports/bifrost-http/lib"
+	"github.com/maximhq/bifrost/transports/bifrost-http/tracking"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/valyala/fasthttp"
+	"github.com/valyala/fasthttp/fasthttpadaptor"
 )
 
 // Command line flags
 var (
-	initialPoolSize    int    // Initial size of the connection pool
-	dropExcessRequests bool   // Drop excess requests
-	port               string // Port to run the server on
-	configPath         string // Path to the config file
-	envPath            string // Path to the .env file
+	initialPoolSize    int      // Initial size of the connection pool
+	dropExcessRequests bool     // Drop excess requests
+	port               string   // Port to run the server on
+	configPath         string   // Path to the config file
+	envPath            string   // Path to the .env file
+	prometheusLabels   []string // Labels to add to Prometheus metrics (optional)
```

Comment:
@akshaydeo user can pass custom labels from here, which will be caught from the headers